NPE in MultiplexBluetoothTransport

I am seeing NullPointerException from [here](https://github.com/smartdevicelink/sdl_android/blob/master/sdl_android/src/main/java/com/smartdevicelink/transport/MultiplexBluetoothTransport.java#L180)
in MultiplexBluetoothTransport.  It seems like this line in the code is doing a null check on an instance of a static variable so I'm guessing that the static variable itself is null.

Here is the stack trace:
```
07-11 10:22:24.014 15384 15384 E AndroidRuntime: java.lang.RuntimeException: Unable to start service com.main.tcapp.sdl.SdlRouterService@2052c2cb with Intent { cmp=com.toyota.tcapp/com.main.tcapp.sdl.SdlRouterService (has extras) }: java.lang.NullPointerException: Attempt to read from field 'com.smartdevicelink.transport.MultiplexBluetoothTransport$ConnectThread com.smartdevicelink.transport.MultiplexBluetoothTransport.mConnectThread' on a null object reference
07-11 10:22:24.014 15384 15384 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:2881)
07-11 10:22:24.014 15384 15384 E AndroidRuntime: 	at android.app.ActivityThread.access$2100(ActivityThread.java:144)
07-11 10:22:24.014 15384 15384 E AndroidRuntime: 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1376)
07-11 10:22:24.014 15384 15384 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:102)
07-11 10:22:24.014 15384 15384 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:135)
07-11 10:22:24.014 15384 15384 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:5221)
07-11 10:22:24.014 15384 15384 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
07-11 10:22:24.014 15384 15384 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Method.java:372)
07-11 10:22:24.014 15384 15384 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:899)
07-11 10:22:24.014 15384 15384 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:694)
07-11 10:22:24.014 15384 15384 E AndroidRuntime: Caused by: java.lang.NullPointerException: Attempt to read from field 'com.smartdevicelink.transport.MultiplexBluetoothTransport$ConnectThread com.smartdevicelink.transport.MultiplexBluetoothTransport.mConnectThread' on a null object reference
07-11 10:22:24.014 15384 15384 E AndroidRuntime: 	at com.smartdevicelink.transport.MultiplexBluetoothTransport.start(MultiplexBluetoothTransport.java:180)
07-11 10:22:24.014 15384 15384 E AndroidRuntime: 	at com.smartdevicelink.transport.SdlRouterService.initBluetoothSerialService(SdlRouterService.java:1132)
07-11 10:22:24.014 15384 15384 E AndroidRuntime: 	at com.smartdevicelink.transport.SdlRouterService.onStartCommand(SdlRouterService.java:899)
07-11 10:22:24.014 15384 15384 E AndroidRuntime: 	at com.main.tcapp.sdl.SdlRouterService.onStartCommand(SdlRouterService.java:237)
07-11 10:22:24.014 15384 15384 E AndroidRuntime: 	at android.app.ActivityThread.handleServiceArgs(ActivityThread.java:2864)
07-11 10:22:24.014 15384 15384 E AndroidRuntime: 	... 9 more
```

I wonder if it is due to the change that went in to fix an earlier [issue](https://github.com/smartdevicelink/sdl_android/pull/537/commits/c30dd431c8767bd772ad3139386c623846c5ba9a).   However, it happens on starting the SDL Router service.

I'm using master plus cherry-picked the commit for that fix.

--- a/sdl_android/src/androidTest/java/com/smartdevicelink/transport/SdlRouterServiceTests.java
+++ b/sdl_android/src/androidTest/java/com/smartdevicelink/transport/SdlRouterServiceTests.java
@@ -108,7 +108,7 @@ public class SdlRouterServiceTests extends AndroidTestCase {
 
             //Send a non-null bundle with a null bytes array
             //First, set mSerialService to the correct state so we get to test packet being null
-            MultiplexBluetoothTransport transport = MultiplexBluetoothTransport.getBluetoothSerialServerInstance(null);
+            MultiplexBluetoothTransport transport = new MultiplexBluetoothTransport(null);
             transport.setStateManually(MultiplexBluetoothTransport.STATE_CONNECTED);
             field = SdlRouterService.class.getDeclaredField("mSerialService");
             field.setAccessible(true);

--- a/sdl_android/src/main/java/com/smartdevicelink/transport/MultiplexBluetoothTransport.java
+++ b/sdl_android/src/main/java/com/smartdevicelink/transport/MultiplexBluetoothTransport.java
@@ -50,10 +50,14 @@ public class MultiplexBluetoothTransport {
     private static final UUID SERVER_UUID= new UUID(0x936DA01F9ABD4D9DL, 0x80C702AF85C822A8L);
     // Name for the SDP record when creating server socket
     private static final String NAME_SECURE =" SdlRouterService";
-    
+    // Key names received from the BluetoothSerialServer Handler
+    public static final String DEVICE_NAME = "device_name";
+    public static final String TOAST = "toast";
+    private static final long MS_TILL_TIMEOUT = 2500;
+    private static final int READ_BUFFER_SIZE = 4096;
+    private static final Object THREAD_LOCK =  new Object();;
 
-	protected static final String SHARED_PREFS = "sdl.bluetoothprefs";
-
+    protected static final String SHARED_PREFS = "sdl.bluetoothprefs";
 
     // Constants that indicate the current connection state
     public static final int STATE_NONE 			= 0;    // we're doing nothing
@@ -67,40 +71,24 @@ public class MultiplexBluetoothTransport {
     private final BluetoothAdapter mAdapter = BluetoothAdapter.getDefaultAdapter();
     private final Handler mHandler;
     private AcceptThread mSecureAcceptThread; 
-    private static Object threadLock = null;
     private ConnectThread mConnectThread;
-    private static ConnectedThread mConnectedThread; //I HATE ALL THIS STATIC CRAP, But it seems like the only way right now
-    private static ConnectedWriteThread mConnectedWriteThread; //I HATE ALL THIS STATIC CRAP, But it seems like the only way right now
-
-    private  static int mState;
-    
-    // Key names received from the BluetoothSerialServer Handler
-    public static final String DEVICE_NAME = "device_name";
-    public static final String TOAST = "toast";
-    
-    
+    private ConnectedThread mConnectedThread;
+    private ConnectedWriteThread mConnectedWriteThread;
+    private int mState;
     private int mBluetoothLevel = 0;
     Handler timeOutHandler;
     Runnable socketRunable;
-    private static final long msTillTimeout = 2500;
-
-    private static final int READ_BUFFER_SIZE = 4096;
+    boolean keepSocketAlive = true;
 
     public static String currentlyConnectedDevice = null;
     public static String currentlyConnectedDeviceAddress = null;
-    private static MultiplexBluetoothTransport serverInstance = null;
-    //private BluetoothServerSocket serverSocket= null;
 
 
-	static boolean keepSocketAlive = true;
-    
-    
     /**
      * Constructor. Prepares a new BluetoothChat session.
-     * @param context  The UI Activity Context
      * @param handler  A Handler to send messages back to the UI Activity
      */
-    private MultiplexBluetoothTransport(Handler handler) {
+    public MultiplexBluetoothTransport(Handler handler) {
     	//Log.w(TAG, "Creating Bluetooth Serial Adapter");
        // mAdapter = BluetoothAdapter.getDefaultAdapter();
         mState = STATE_NONE;
@@ -108,32 +96,41 @@ public class MultiplexBluetoothTransport {
           
         //This will keep track of which method worked last night
         mBluetoothLevel = SdlRouterService.getBluetoothPrefs(SHARED_PREFS);
-        Object object = new Object();
-        threadLock = object;
-
     }
 
 
-
-	/*
-     * Let's use this method from now on to get bluetooth service
+    /**
+     * This method has been deprecated. It will return an instance of MultiplexBluetoothTransport but <b>should not</b> be used.
+     * @param handler for receiving status messages from the transport
+     * @return an instance of MultiplexBluetoothTransport
+     * @deprecated
      */
+	@Deprecated
     public synchronized static MultiplexBluetoothTransport getBluetoothSerialServerInstance(Handler handler){
-    	return getBluetoothSerialServerInstance(handler,true);
+    	return new MultiplexBluetoothTransport(handler);
     }
-	/*
-     * Let's use this method from now on to get bluetooth service
+    /**
+     * This method has been deprecated. It will return an instance of MultiplexBluetoothTransport but <b>should not</b> be used.
+     * @param handler for receiving status messages from the transport
+     * @param keepSocketAlive Flag for keeping the socket alive
+     * @return an instance of MultiplexBluetoothTransport
+     * @deprecated
      */
+	@Deprecated
     public synchronized static MultiplexBluetoothTransport getBluetoothSerialServerInstance(Handler handler, boolean keepSocketAlive){
-
-        if(serverInstance==null){
-        	serverInstance = new MultiplexBluetoothTransport(handler);
-        }
-        MultiplexBluetoothTransport.keepSocketAlive = keepSocketAlive;
-    	return serverInstance;
+        MultiplexBluetoothTransport transport = new MultiplexBluetoothTransport(handler);
+        transport.setKeepSocketAlive(keepSocketAlive);
+    	return transport;
     }
+
+    /**
+     * This method has been deprecated. <b>It will always return null.</b>
+     * @return always null
+     * @deprecated
+     */
+    @Deprecated
     public synchronized static MultiplexBluetoothTransport getBluetoothSerialServerInstance(){
-    	return serverInstance;
+    	return null;
     }
 
     //These methods are used so we can have a semi-static reference to the Accept Thread (Static reference inherited by housing class)
@@ -167,7 +164,7 @@ public class MultiplexBluetoothTransport {
 
     private void clearInstanceOnError(int state){
         if(state == STATE_ERROR){
-            serverInstance = null;
+           // serverInstance = null;
         }
     }
 
@@ -178,7 +175,7 @@ public class MultiplexBluetoothTransport {
     }
 
     public void setKeepSocketAlive(boolean keepSocketAlive){
-    	MultiplexBluetoothTransport.keepSocketAlive = keepSocketAlive;
+    	this.keepSocketAlive = keepSocketAlive;
     }
     
     /**
@@ -187,7 +184,7 @@ public class MultiplexBluetoothTransport {
     public synchronized void start() {
     	//Log.d(TAG, "Starting up Bluetooth Server to Listen");
         // Cancel any thread attempting to make a connection
-        if (serverInstance.mConnectThread != null) {serverInstance.mConnectThread.cancel(); serverInstance.mConnectThread = null;}
+        if (mConnectThread != null) {mConnectThread.cancel(); mConnectThread = null;}
 
         // Cancel any thread currently running a connection
         if (mConnectedThread != null) {mConnectedThread.cancel(); mConnectedThread = null;}
@@ -196,14 +193,14 @@ public class MultiplexBluetoothTransport {
        
 
         // Start the thread to listen on a BluetoothServerSocket
-        if (getBluetoothSerialServerInstance().getAcceptThread() == null
-        		&& serverInstance.mAdapter != null
-        		&&  serverInstance.mAdapter.isEnabled()) {
+        if (getAcceptThread() == null
+        		&& mAdapter != null
+        		&&  mAdapter.isEnabled()) {
         	//Log.d(TAG, "Secure thread was null, attempting to create new");
-        	getBluetoothSerialServerInstance().setAcceptThread(new AcceptThread(true));
-            if(getBluetoothSerialServerInstance().getAcceptThread()!=null){
-            	getBluetoothSerialServerInstance().setState(STATE_LISTEN);
-            	 getBluetoothSerialServerInstance().getAcceptThread().start();
+        	setAcceptThread(new AcceptThread(true));
+            if(getAcceptThread()!=null){
+            	setState(STATE_LISTEN);
+            	 getAcceptThread().start();
             }
         }
     }
@@ -215,7 +212,7 @@ public class MultiplexBluetoothTransport {
     public synchronized void connect(BluetoothDevice device) {
         // Cancel any thread attempting to make a connection
         if (mState == STATE_CONNECTING) {
-            if (serverInstance.mConnectThread != null) {serverInstance.mConnectThread.cancel(); serverInstance.mConnectThread = null;}
+            if (mConnectThread != null) {mConnectThread.cancel(); mConnectThread = null;}
         }
 
         // Cancel any thread currently running a connection
@@ -230,9 +227,9 @@ public class MultiplexBluetoothTransport {
         }
         
         // Start the thread to connect with the given device
-       serverInstance.mConnectThread = new ConnectThread(device);
-       serverInstance.mConnectThread.start();
-       serverInstance.setState(STATE_CONNECTING);
+       mConnectThread = new ConnectThread(device);
+       mConnectThread.start();
+       setState(STATE_CONNECTING);
     }
 
     /**
@@ -242,9 +239,9 @@ public class MultiplexBluetoothTransport {
      */
     public synchronized void connected(BluetoothSocket socket, BluetoothDevice device) {
         // Cancel the thread that completed the connection
-        if (getBluetoothSerialServerInstance().mConnectThread != null) {
-        	getBluetoothSerialServerInstance().mConnectThread.cancel(); 
-        	getBluetoothSerialServerInstance().mConnectThread = null;
+        if (mConnectThread != null) {
+        	mConnectThread.cancel();
+        	mConnectThread = null;
         }
         
         // Cancel any thread currently running a connection
@@ -257,9 +254,9 @@ public class MultiplexBluetoothTransport {
         	mConnectedWriteThread = null;
         }
         // Cancel the accept thread because we only want to connect to one device
-        if (!keepSocketAlive && getBluetoothSerialServerInstance().mSecureAcceptThread != null) {
-        	getBluetoothSerialServerInstance().mSecureAcceptThread.cancel();
-            getBluetoothSerialServerInstance().mSecureAcceptThread = null;
+        if (!keepSocketAlive && mSecureAcceptThread != null) {
+        	mSecureAcceptThread.cancel();
+            mSecureAcceptThread = null;
         }
 
         // Start the thread to manage the connection and perform transmissions
@@ -280,21 +277,21 @@ public class MultiplexBluetoothTransport {
         Bundle bundle = new Bundle();
         bundle.putString(DEVICE_NAME, device.getName());
         msg.setData(bundle);
-        getBluetoothSerialServerInstance().mHandler.sendMessage(msg);
-        getBluetoothSerialServerInstance().setState(STATE_CONNECTED);
+        mHandler.sendMessage(msg);
+        setState(STATE_CONNECTED);
     }
 
     /**
      * Stop all threads
      */
     public synchronized void stop() {
-    	getBluetoothSerialServerInstance().stop(STATE_NONE);
+    	stop(STATE_NONE);
     }
     protected synchronized void stop(int stateToTransitionTo) {
     	//Log.d(TAG, "Attempting to close the bluetooth serial server");
-        if (getBluetoothSerialServerInstance().mConnectThread != null) {
-        	getBluetoothSerialServerInstance().mConnectThread.cancel(); 
-        	getBluetoothSerialServerInstance().mConnectThread = null;
+        if (mConnectThread != null) {
+        	mConnectThread.cancel();
+        	mConnectThread = null;
         }
 
         if (mConnectedThread != null) {
@@ -308,7 +305,7 @@ public class MultiplexBluetoothTransport {
             mSecureAcceptThread = null;
         }
 
-        getBluetoothSerialServerInstance().setState(stateToTransitionTo);
+        setState(stateToTransitionTo);
     }
     
 
@@ -339,7 +336,7 @@ public class MultiplexBluetoothTransport {
         Bundle bundle = new Bundle();
         bundle.putString(TOAST, "Unable to connect device");
         msg.setData(bundle);
-        getBluetoothSerialServerInstance().mHandler.sendMessage(msg);
+        mHandler.sendMessage(msg);
 
         // Start the service over to restart listening mode
        // BluetoothSerialServer.this.start();
@@ -354,14 +351,14 @@ public class MultiplexBluetoothTransport {
         Bundle bundle = new Bundle();
         bundle.putString(TOAST, "Device connection was lost");
         msg.setData(bundle);
-        getBluetoothSerialServerInstance().mHandler.sendMessage(msg);
-        getBluetoothSerialServerInstance().stop();
+        mHandler.sendMessage(msg);
+        stop();
 
     }
     
     private void timerDelayRemoveDialog(final BluetoothSocket sock){
-    	getBluetoothSerialServerInstance().timeOutHandler = new Handler(); 
-    	getBluetoothSerialServerInstance().socketRunable = new Runnable() {           
+    	timeOutHandler = new Handler();
+    	socketRunable = new Runnable() {
             public void run() {
             	//Log.e(TAG, "BLUETOOTH SOCKET CONNECT TIMEOUT - ATTEMPT TO CLOSE SOCKET");
             	try {
@@ -371,7 +368,7 @@ public class MultiplexBluetoothTransport {
 				}         
             }
         };
-        getBluetoothSerialServerInstance().timeOutHandler.postDelayed(socketRunable, msTillTimeout); 
+        timeOutHandler.postDelayed(socketRunable, MS_TILL_TIMEOUT);
     }
     
     
@@ -388,14 +385,14 @@ public class MultiplexBluetoothTransport {
         
         @SuppressLint("NewApi")
 		public AcceptThread(boolean secure) {
-        	synchronized(threadLock){
+        	synchronized(THREAD_LOCK){
             	//Log.d(TAG, "Creating an Accept Thread");
             	BluetoothServerSocket tmp = null;
             	mSocketType = secure ? "Secure":"Insecure";
             	// Create a new listening server socket
             	try {
                 	if (secure) {
-                		tmp = getBluetoothSerialServerInstance().mAdapter.listenUsingRfcommWithServiceRecord(NAME_SECURE, SERVER_UUID);
+                		tmp = mAdapter.listenUsingRfcommWithServiceRecord(NAME_SECURE, SERVER_UUID);
                 	}
             	} catch (IOException e) {
                 	//Log.e(TAG, "Socket Type: " + mSocketType + "listen() failed", e);
@@ -412,7 +409,7 @@ public class MultiplexBluetoothTransport {
         }
         
         public void run() {
-            synchronized(threadLock){
+            synchronized(THREAD_LOCK){
             	Log.d(TAG, "Socket Type: " + mSocketType +
                     " BEGIN mAcceptThread" + this);
             setName("AcceptThread" + mSocketType);
@@ -425,7 +422,7 @@ public class MultiplexBluetoothTransport {
                 try {
                 	if(listenAttempts>=5){
                 		Log.e(TAG, "Complete failure in attempting to listen for Bluetooth connection, erroring out.");
-                		getBluetoothSerialServerInstance().stop(STATE_ERROR);
+                		MultiplexBluetoothTransport.this.stop(STATE_ERROR);
                 		return;
                 	}
                 	listenAttempts++;
@@ -440,12 +437,12 @@ public class MultiplexBluetoothTransport {
                 	}
                 	else{
                 		Log.e(TAG, "Listening Socket was null, stopping the bluetooth serial server.");
-                		getBluetoothSerialServerInstance().stop(STATE_ERROR);
+                		MultiplexBluetoothTransport.this.stop(STATE_ERROR);
                 		return;
                 	}
                 } catch (IOException e) {
                     Log.e(TAG, "Socket Type: " + mSocketType + "accept() failed");
-                    getBluetoothSerialServerInstance().stop(STATE_ERROR);
+                    MultiplexBluetoothTransport.this.stop(STATE_ERROR);
                     return;
                 }
 
@@ -456,7 +453,7 @@ public class MultiplexBluetoothTransport {
                         case STATE_LISTEN:
                         case STATE_CONNECTING:
                             // Situation normal. Start the connected thread.
-                        	getBluetoothSerialServerInstance().connected(socket, socket.getRemoteDevice());
+                        	connected(socket, socket.getRemoteDevice());
                             
                             break;
                         case STATE_NONE:
@@ -739,7 +736,7 @@ public class MultiplexBluetoothTransport {
                 // Exception during write
             	//OMG! WE MUST NOT BE CONNECTED ANYMORE! LET THE USER KNOW
             	Log.e(TAG, "Error sending bytes to connected device!");
-            	getBluetoothSerialServerInstance().connectionLost();
+            	connectionLost();
             }
         }
 

--- a/sdl_android/src/main/java/com/smartdevicelink/transport/MultiplexBluetoothTransport.java
+++ b/sdl_android/src/main/java/com/smartdevicelink/transport/MultiplexBluetoothTransport.java
@@ -450,9 +450,7 @@ public class MultiplexBluetoothTransport {
                             // Either not ready or already connected. Terminate new socket.
                             try {
                             	Log.d(TAG, "Close unwanted socket");
-                                if(socket!=null){
-                                	socket.close();
-                                }
+                               	socket.close();
                             } catch (IOException e) {
                                 Log.e(TAG, "Could not close unwanted socket", e);
                             }

--- a/sdl_android/src/main/java/com/smartdevicelink/transport/MultiplexBluetoothTransport.java
+++ b/sdl_android/src/main/java/com/smartdevicelink/transport/MultiplexBluetoothTransport.java
@@ -256,16 +256,15 @@ public class MultiplexBluetoothTransport {
         mConnectedWriteThread = new ConnectedWriteThread(socket);
         mConnectedWriteThread.start();
 
-        //Store a static name of the device that is connected. We can do this since the only time
-        //we will access it will be when we receive a CONNECT packet from a device
-        if(device!=null && device.getName()!=null && device.getName()!=""){
+        //Store a static name of the device that is connected.
+        if(device!=null){
         	currentlyConnectedDevice = device.getName();
         }
         
         // Send the name of the connected device back to the UI Activity
         Message msg = mHandler.obtainMessage(SdlRouterService.MESSAGE_DEVICE_NAME);
         Bundle bundle = new Bundle();
-        bundle.putString(DEVICE_NAME, device.getName());
+        bundle.putString(DEVICE_NAME, currentlyConnectedDevice);
         msg.setData(bundle);
         mHandler.sendMessage(msg);
         setState(STATE_CONNECTED);

--- a/sdl_android/src/main/java/com/smartdevicelink/transport/MultiplexBluetoothTransport.java
+++ b/sdl_android/src/main/java/com/smartdevicelink/transport/MultiplexBluetoothTransport.java
@@ -143,8 +143,6 @@ public class MultiplexBluetoothTransport {
     protected synchronized void setStateManually(int state){
         //Log.d(TAG, "Setting state from: " +mState + " to: " +state);
         mState = state;
-
-        clearInstanceOnError(state);
     }
     /**
      * Set the current state of the chat connection
@@ -158,14 +156,6 @@ public class MultiplexBluetoothTransport {
         // Give the new state to the Handler so the UI Activity can update
         //Also sending the previous state so we know if we lost a connection
         mHandler.obtainMessage(SdlRouterService.MESSAGE_STATE_CHANGE, state, previousState).sendToTarget();
-
-        clearInstanceOnError(state);
-    }
-
-    private void clearInstanceOnError(int state){
-        if(state == STATE_ERROR){
-           // serverInstance = null;
-        }
     }
 
     /**

--- a/sdl_android/src/main/java/com/smartdevicelink/transport/SdlRouterService.java
+++ b/sdl_android/src/main/java/com/smartdevicelink/transport/SdlRouterService.java
@@ -109,7 +109,7 @@ public class SdlRouterService extends Service{
 	
     private final int UNREGISTER_APP_INTERFACE_CORRELATION_ID = 65530;
     
-	private static MultiplexBluetoothTransport mSerialService = null;
+	private MultiplexBluetoothTransport mSerialService = null;
 
 	private static boolean connectAsClient = false;
 	private static boolean closing = false;
@@ -196,16 +196,8 @@ public class SdlRouterService extends Service{
 				Log.e(TAG, "Serial service not initliazed while registering app");
 				//Maybe we should try to do a connect here instead
 				Log.d(TAG, "Serial service being restarted");
-				if(mSerialService ==null){
-					Log.e(TAG, "Local copy of BT Server is null");
-					mSerialService = MultiplexBluetoothTransport.getBluetoothSerialServerInstance();
-					if(mSerialService==null){
-						Log.e(TAG, "Local copy of BT Server is still null and so is global");
-						mSerialService = MultiplexBluetoothTransport.getBluetoothSerialServerInstance(mHandlerBT);
+				initBluetoothSerialService();
 
-					}
-				}
-				mSerialService.start();
 
 			}
 		}
@@ -1121,18 +1113,16 @@ public class SdlRouterService extends Service{
 			Log.d(TAG, "Not starting own bluetooth during legacy mode");
 			return;
 		}
-		Log.i(TAG, "Iniitializing bluetooth transport");
 		//init serial service
-		if(mSerialService ==null){
-			mSerialService = MultiplexBluetoothTransport.getBluetoothSerialServerInstance();
-			if(mSerialService==null){
-				mSerialService = MultiplexBluetoothTransport.getBluetoothSerialServerInstance(mHandlerBT);
-			}
+		if(mSerialService == null || mSerialService.getState() == MultiplexBluetoothTransport.STATE_ERROR){
+			Log.i(TAG, "Initializing bluetooth transport");
+			mSerialService = new MultiplexBluetoothTransport(mHandlerBT);
 		}
 		if (mSerialService != null) {
             // Only if the state is STATE_NONE, do we know that we haven't started already
-            if (mSerialService.getState() == MultiplexBluetoothTransport.STATE_NONE || mSerialService.getState() == MultiplexBluetoothTransport.STATE_ERROR) {
+            if (mSerialService.getState() == MultiplexBluetoothTransport.STATE_NONE) {
               // Start the Bluetooth services
+				Log.i(TAG, "Starting bluetooth transport");
             	mSerialService.start();
             }
 
@@ -1580,7 +1570,7 @@ public class SdlRouterService extends Service{
     		Log.d(TAG,"Connecting to device: " + device.getName().toString());
 			if(mSerialService == null || !mSerialService.isConnected())
 			{	// Set up the Bluetooth serial object				
-				mSerialService = MultiplexBluetoothTransport.getBluetoothSerialServerInstance(mHandlerBT);
+				mSerialService = new MultiplexBluetoothTransport(mHandlerBT);
 			}
 			// We've been given a device - let's connect to it
 			if(mSerialService.getState()!=MultiplexBluetoothTransport.STATE_CONNECTING){//mSerialService.stop();
@@ -1673,7 +1663,7 @@ public class SdlRouterService extends Service{
             	//Log.v(TAG, "Self service info " + self);
             	//Log.v(TAG, "Newest compare to service info " + newestServiceReceived);
             	if(newestServiceReceived!=null && self.isNewer(newestServiceReceived)){
-            		if(SdlRouterService.mSerialService!=null && SdlRouterService.mSerialService.isConnected()){ //We are currently connected. Wait for next connection 
+            		if(SdlRouterService.this.mSerialService!=null && SdlRouterService.this.mSerialService.isConnected()){ //We are currently connected. Wait for next connection
             			return;
             		}
             		Log.d(TAG, "There is a newer version "+newestServiceReceived.version+" of the Router Service, starting it up");

